### PR TITLE
boards: stm32h750b_dk: update doc & board yaml file

### PR DESCRIPTION
**Pull request by JarmouniA**
Update STM32H750B-DK doc & `yaml` file with enabled features.

### CODE REVIEW

File: boards/st/stm32h750b_dk/doc/index.rst [hunk 1]
```diff
@@ -59,7 +59,14 @@ The current Zephyr stm32h750b_dk board configuration supports the following hard
 +-----------+------------+-------------------------------------+
 | RTC       | on-chip    | rtc                                 |
 +-----------+------------+-------------------------------------+
-
+| ADC       | on-chip    | adc                                 |
```
Comment (erwango) [inline boards/st/stm32h750b_dk/doc/index.rst:62]: ADC is not enabled in dts

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwsie LGTM

File: boards/st/stm32h750b_dk/doc/index.rst (hunk 1 above)
Comment (erwango) [reply boards/st/stm32h750b_dk/doc/index.rst:62]: This is valid indeed.

**Review by erwango — APPROVED** [review]

File: boards/st/stm32h750b_dk/stm32h750b_dk.yaml [hunk 2]
```diff
@@ -15,4 +15,6 @@ supported:
   - flash
   - rtc
   - memc
+  - display
+  - spi
```
Comment (FRASTM) [inline boards/st/stm32h750b_dk/stm32h750b_dk.yaml:19]: maybe also 
 ` - qspi`

Comment (JarmouniA) [reply boards/st/stm32h750b_dk/stm32h750b_dk.yaml:19]: @FRASTM There is no test/test case in-tree that uses the `qspi` tag & concerns this board.

**Review by FRASTM — APPROVED** [review]

**Pull request merged**